File: csharp_3_tests/GildedRose/GildedRoseTest.cs
#Copyright (c) 2023, Paul Moore
#All rights reserved.
#
#This source code is licensed under the BSD-style license found in the
#LICENSE file in the root directory of this source tree. 

using NUnit.Framework;
using System.Collections.Generic;

namespace GildedRose
{
    [TestFixture]
    public class GildedRoseTest
    {
        private const string BackstagePass = "Backstage passes to a TAFKAL80ETC concert";

        private List<Item> createItemList(string itemName, int sellIn, int quality) {
            return new List<Item> { new Item { Name = itemName, SellIn = sellIn, Quality = quality } };
        }        

        [Test]
        public void AgedBrie_QualityIncreases_EvenAfterSellInDate() {
            GildedRose sut = new GildedRose(createItemList("Aged Brie", -1, 20));
            sut.UpdateQuality();
            Assert.AreEqual(22, sut.Items[0].Quality, "Aged Brie improves twice as fast after sellin date (BUG?)");
        }

        [Test]
        public void BackStagePass_QualityDropsToZeroWhenConcertPasses() {
            GildedRose sut = new GildedRose(createItemList(BackstagePass, 0, 40));
            sut.UpdateQuality();
            Assert.AreEqual(0, sut.Items[0].Quality, "Backstage Pass is worthless when concert has passed");
        }
        
        [Test]
        public void GenericItem_QualityDecreasesTwiceAsFastAfterSellinDate() {
            GildedRose sut = new GildedRose(createItemList("generic item", 0, 10));
            sut.UpdateQuality();
            Assert.AreEqual(8, sut.Items[0].Quality, "When sellin date is 0 then quality decreases twice as fast");
        }
    
    }
}

File: java_0_tests/src/test/java/gildedrose/GildedRoseTest.java
//Copyright (c) 2023, Paul Moore
//All rights reserved.
//
//This source code is licensed under the BSD-style license found in the
//LICENSE file in the root directory of this source tree.

package gildedrose;

import org.junit.jupiter.api.Disabled;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

public class GildedRoseTest {

    public static final String BACKSTAGE_PASS = "Backstage passes to a TAFKAL80ETC concert";

    @Test
    @Disabled
    void AgedBrie_QualityIncreases_EvenAfterSellInDate() {
        GildedRose sut = new GildedRose(createItemArray("Aged Brie", -1, 20));

        sut.updateQuality();

        assertEquals(22, sut.items[0].quality, "Aged Brie improves twice as fast after sellIn date (BUG?)");
    }

    @Test
    @Disabled
    void BackStagePass_QualityDropsToZeroWhenConcertPasses() {
        GildedRose sut = new GildedRose(createItemArray(BACKSTAGE_PASS, 0, 40));

        sut.updateQuality();

        assertEquals(0, sut.items[0].quality, "Backstage Pass is worthless when concert has passed");
    }

    @Test
    @Disabled
    void GenericItem_QualityDecreasesTwiceAsFastAfterSellInDate() {
        GildedRose sut = new GildedRose(createItemArray("generic item", 0, 10));

        sut.updateQuality();

        assertEquals(8, sut.items[0].quality, "When sellIn date is 0 then quality decreases twice as fast");
    }

    private Item[] createItemArray(String itemName, int sellIn, int quality) {
        return new Item[] { new Item(itemName, sellIn, quality) };
    }
}
